examples/molecular_dynamics/lennard_jones_system.py

```diff
@@ -0,0 +1,280 @@
+                log.log_minibatch({"Mini-batch loss": loss.detach()})
+            log.log_epoch({"Learning Rate": optimizer.param_groups[0]["lr"]})
+
+        if dist.rank == 0:
+            save_checkpoint(
+                "./checkpoints",
+                models=model,
+                optimizer=optimizer,
+                scheduler=scheduler,
+                epoch=epoch,
+            )
+
+            with LaunchLogger("valid", epoch=epoch) as log:
+                with torch.no_grad():
+                    model.eval()
+                    val_loss = 0
+
+                    forces_pair = []
+                    cosines = []
+                    for data in test_dataloader:
+
+                        pos = data[0][0]
+                        forces = data[1][0]
+                        pos, forces = prepare_input(
+                            pos,
+                            forces,
+                            box_size,
+                            rotation_matrix=None,
+                            add_random_noise=False,
+                        )
+
+                        # Create edges
+                        src, dst, edge_features = create_edges(
+                            pos, distance_threshold, box_size
+                        )
+                        g = dgl.graph((src, dst)).to(dist.device)
+                        node_fea = torch.ones(
+                            size=(pos.shape[0], cfg.model.hidden_dim_edge_encoder)
+                        ).to(dist.device)
+                        edge_fea = (
+                            torch.tensor(np.array(edge_features), dtype=torch.float32)
+                            .view(-1, 4)
+                            .to(dist.device)
+                        )
+
+                        out = model(node_fea, edge_fea, g)
+                        true_out = torch.tensor(
+                            (forces - force_mean) / force_sd, dtype=torch.float32
+                        ).to(dist.device)
+
+                        val_loss += F.mse_loss(out, true_out).detach()
+
+                        out_np = out.detach().cpu().numpy()
+                        true_out_np = true_out.detach().cpu().numpy()
+                        forces_pair.append((out_np, true_out_np))
+
+                        # angle computation
+                        dot_product = np.sum(out_np * true_out_np, axis=1)
+                        out_np_mag = np.linalg.norm(out_np, axis=1)
+                        true_out_np_mag = np.linalg.norm(true_out_np, axis=1)
+
+                        cosine = dot_product / (out_np_mag * true_out_np_mag)
+                        cosines.append(cosine)
+
+                    plt.clf()
+                    plt.figure(figsize=(5, 5))
+                    for force_system in forces_pair:
+                        pred, true = force_system
+                        pred_mag = pred[:, 0] + pred[:, 1] + pred[:, 2]
+                        true_mag = true[:, 0] + true[:, 1] + true[:, 2]
```

Comment:
Again, can we have commentary on this? I am not sure how useful it is to sum the unsigned force components. Wouldn't I want the magnitude instead?

Reply:
Good point. Magnitude makes sense to me too, but the publication plot shows +ve and -ve components indicating that it is the total force. Plotting vector quantities on a 2d plot is not the best way, but I wanted to keep an easy way to reproduce plots